runtime/drivers/clickhouse/embed.go

```diff
@@ -115,7 +116,31 @@ func (e *embedClickHouse) start() (*clickhouse.Options, error) {
 	// We read StderrPipe initially to check for clickhouse running status.
 	// Once the process is closed the stderr pipe will be closed too, io.Copy will return EOF and the goroutine will exit.
 	go func() {
-		_, _ = io.Copy(io.Discard, stderr)
+		decoder := json.NewDecoder(stderr)
+
+		for {
+			var log clickhouseLog
+			if err := decoder.Decode(&log); err != nil {
+				if err == io.EOF {
+					// EOF means the process has exited and the stderr pipe is closed.
+					break
+				}
+				e.logger.Error("Failed to decode ClickHouse log", zap.Error(err))
+			}
+
+			switch log.Level {
+			case "Fatal":
+				e.logger.Fatal("ClickHouse embedded server: fatal log received, restart server", zap.String("logger_name", log.LoggerName), zap.String("message", log.Message))
```

Comment:
On the Zap side, should we map this to `error` level? We generally avoid fatal everywhere since it will not even run deferred code in any upstream logic.